### ISSUE TITLE
Suggestion: Add documentation

### ISSUE THREAD

**Issue by chris13524** [Apostolique/Agar.io-bot#99]
The lines and thing are mostly intuitive, but it would be nice to have a full set of documentation describing what the lines do, what they mean, and how the bot reacts. I know this is very time consuming, but might be nice to do in your free time.

**Comment by chris13524** [comment]
Like I have trouble understanding in the new version what all the circles around the blobs do. I would assume that they are possible attach areas and the bot should avoid/get out of them. But I am confused about the orange zones around viruses and the red one around the player himself.

**Comment by Apostolique** [comment]
The lines are mostly for debugging purposes, they change every time I feel like it. The circles right now are the splitting range. If you're inside a red circle, the enemy can split on you for sure. The orange circles are the threat circles of enemies that cannot split, the current size is just a political choice.

You can also see [red - orange] arcs around the player, they show which enemies are marked as current threats. The green arcs are just areas which have the property of never pointing towards an enemy, if you travel towards green areas, you should be safe. In the current version though, green areas are not computed correctly which means the bot will sometimes run into enemies trying to go towards the green.

**Comment by chris13524** [comment]
I like the lines, the bot isn't very interesting (to me) without them, just a blob moving around the screen. I think they should be more than a debugging feature, but that's just me. And why do the green arcs disappear when there are no threats? Seems like an inconsistency to me. What about the giant pink circle around the player? What does that do?

**Comment by chris13524** [comment]
And what are those red dots that look like they are being shot out of the player? At some points they show the outer bounds of an enemy, but at other times, they just float around like they are being shot.

**Comment by Apostolique** [comment]
Of course, I keep the good looking lines. The current version looks pretty good for example. And I pay good attention to the way that they are rendered.

I've updated the red dot to be better, pushing next version right now.

**Comment by chris13524** [comment]
Yes, but what is the red line around the player do? My curiosity is killing me!!! And are the tiny yellow circles randomly around the map possible target points to track to?

**Comment by Apostolique** [comment]
I responded to that earlier, and a single yellow dot is just a group of food. Instead of taking all the dots one by one, you can just travel to the middle of them.

**Comment by FliiFe** [comment]
Related : We should remove the dot under the mouse. It's unusefull (afaik) and annoying. I mean, the user know where his mouse is, right ?

**Comment by ErmiyaEskandary** [comment]
@FliiFe What dot ? the yellow one ?
